### ISSUE TITLE
Physics Interpolation affects Camera2D position smoothing on higher framerates

### ISSUE THREAD

**Issue by TotoShampoin** [godotengine/godot#95869]
### Tested versions

- Reproducible in 4.2.stable, 4.3.stable

### System information

ArchLinux 6.10.2-arch1-1.1-g14 - KDE Plasma 6.1.4 - Godot 4.3.stable.arch_linux

### Issue description

When physics interpolation is activated, the camera smoothing actually goes faster than when it's deactivated for the same speed.

### Steps to reproduce

1. Have a monitor with a refresh rate higher than 60Hz
2. Set up a 2D platformer, with a Camera2D that has Position Smoothing turned on
3. Test the camera's physics
4. Go to settings, turn on physics interpolation
5. Test again

### Minimal reproduction project (MRP)

[test.zip](https://github.com/user-attachments/files/16679694/test.zip)

**Comment by rburing** [comment]
I don't have a monitor with a high refresh rate, but I confirm the issue is also visible when setting e.g. `position_smoothing_speed = 0.1` px/s.

This must be related to `Camera2D::get_camera_transform` getting called internally, which has the (cursed) side-effect of also updating the smoothing. Ideally we would untangle that Camera2D code and fix this bug in the process.

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot/issues/49747.

**Comment by rburing** [comment]
In the MRP here (which sets the Camera2D node position in code) the speedup is caused by the fact that both on `NOTIFICATION_INTERNAL_PYSICS_PROCESS` and on `NOTIFICATION_TRANSFORM_CHANGED` there is a call to `Camera2D::get_camera_transform()` which has the side-effect of updating the smoothing (among other things).

cc @lawnjelly since this also affects 3.x

**Comment by TotoShampoin** [comment]
........... Why should a getter even have a side effect?

**Comment by lawnjelly** [comment]
Yes, I have a feeling this was noticed before, I put a note in the source that the `get_camera_transform()` was a historical bug that needs refactor.

In the case of a smoothed physics interpolated camera, presumably either interpolation needs to be switched off (if using it on idle), or it needs to be used on physics tick and the interpolation take over in between. Trying to use it on idle with physics interpolation won't make sense anyway and there will be jitter (it should be hard coded prevented).
